pages/docs/couple-your-code/couple-your-code-waveform.md

```diff
@@ -0,0 +1,102 @@
+
+### Example for waveform iteration with linear interpolation
+
+Linear interpolation between coupling boundary conditions of the previous and the current time window is illustrated below:
+
+![Coupling data exchange with linear interpolation](images/docs/couple-your-code/couple-your-code-waveform/WaveformLinear.png)
+
+If the Dirichlet participant $$\mathcal{D}$$ calls `readBlockVectorData`, it samples the data from a time-dependent function $$c_\mathcal{D}(t)$$. This function is created from linear interpolation of the first and the last sample $$c_{\mathcal{D}0}$$ and $$c_{\mathcal{D}5}$$ created by the Neumann participant $$\mathcal{N}$$ in the current time window. This allows $$\mathcal{D}$$ to sample the coupling condition at arbitrary times $$t$$ inside the current time window.
+
+## Experimental API for waveform iteration
+
+If we want to improve the accuracy by using waveforms, this requires an extension of the existing API, because we need a way to tell preCICE where we want to sample the waveform. For this purpose, preCICE offers an experimental API. Here, `readBlockVectorData` accepts an additional argument `relativeReadTime`. This allows us to choose the time where the waveform should be sampled:
+
+```cpp
+// stable API with constant data in time window
+void readBlockVectorData(int dataID, int size, const int* valueIndices, double* values) const;
+
+// experimental API for waveform iteration
+void readBlockVectorData(int dataID, int size, const int* valueIndices, double relativeReadTime, double* values) const;
+```
+
+`relativeReadTime` describes the time relatively to the beginning of the current timestep. This means that `relativeReadTime = 0` gives us access to data at the beginning of the timestep. By choosing `relativeReadTime > 0` we can sample data at later points. The maximum allowed `relativeReadTime` corresponds to the remaining time until the end of the current time window. Remember that the remaining time until the end of the time window is always returned when calling `precice_dt = precice.advance(dt)` as `precice_dt`. So `relativeReadTime = precice_dt` corresponds to sampling data at the end of the current time window.
+
+If we choose to use a smaller timestep size `dt < precice_dt`, we apply subcycling and therefore `relativeReadTime = dt` corresponds to sampling data at the end of the timestep. But we can also use smaller values for `relativeReadTime`, as shown in the usage example below. When using subcycling, it is important to note that `relativeReadTime = precice_dt` is the default behavior, if no `relativeReadTime` is provided, because preCICE cannot know the `dt` our solver wants to use. This also means that is subcycling is applied one must use the experimental API and provide `relativeReadTime` to benefit from the higher accuracy waveforms.
```

Comment:
```suggestion
If we choose to use a smaller timestep size `dt < precice_dt`, we apply subcycling and therefore `relativeReadTime = dt` corresponds to sampling data at the end of the timestep. But we can also use smaller values for `relativeReadTime`, as shown in the usage example below. When using subcycling, it is important to note that `relativeReadTime = precice_dt` is the default behavior, if no `relativeReadTime` is provided, because preCICE cannot know the `dt` our solver wants to use. This also means that if subcycling is applied one must use the experimental API and provide `relativeReadTime` to benefit from the higher accuracy waveforms.
```